Please merge latest gopcua library

### Relevant telegraf.conf:
<!-- Place config in the toml code section. -->
```[[inputs.opcua]]
  ## Metric name
  # name = "opcua"
  #
  ## OPC UA Endpoint URL
  endpoint = "opc.tcp://10.79.116.248:4840"
  #
  ## Maximum time allowed to establish a connect to the endpoint.
  connect_timeout = "30s"
  #
  ## Maximum time allowed for a request over the estabilished connection.
  request_timeout = "20s"
  #
  ## Security policy, one of "None", "Basic128Rsa15", "Basic256",
  ## "Basic256Sha256", or "auto"
  security_policy = "None"
  #
  ## Security mode, one of "None", "Sign", "SignAndEncrypt", or "auto"
  security_mode = "None"
  #
  ## Path to cert.pem. Required when security mode or policy isn't "None".
  ## If cert path is not supplied, self-signed cert and key will be generated.
  # certificate = "/etc/telegraf/cert.pem"
  #
  ## Path to private key.pem. Required when security mode or policy isn't "None".
  ## If key path is not supplied, self-signed cert and key will be generated.
  # private_key = "/etc/telegraf/key.pem"
  #
  ## Authentication Method, one of "Certificate", "UserName", or "Anonymous".  To
  ## authenticate using a specific ID, select 'Certificate' or 'UserName'
  auth_method = "Anonymous"
  #
  ## Username. Required for auth_method = "UserName"
  # username = ""
  #
  ## Password. Required for auth_method = "UserName"
  # password = ""
  #
   nodes = [
     {name="CabinetTemp_M1", namespace="2", identifier_type="i", identifier="1030"}
   ]

```

### System info:

<!-- Include Telegraf version, operating system, and other relevant details -->
I'm currently using Telefraf 1.20.0-6f2553d6 for Windows (amd64), running on Windows 10.  I am attempting to connect to a Red Lion DA50 running a Crimson v3.1 OPC UA server.

Telegraf will connect to a Kepware v6.10 and a ProSys OPC UA simulation server.  It will not connect to the Red Lion.  However, I am able to connect to the Red Lion using the gopcua library v0.1.13.

### Steps to reproduce:

<!-- Describe the steps to reproduce the bug. -->

1. If I use the configuration above, I get: Error in plugin: registerNodes failed: The operation timed out. StatusBadTimeout (0x800A0000) when attempting to connect to the Red Lion server using Telegraf.

2. When I connect to the ProSys server, or the KepWare server, the data logs to my local InfluxDB installation using Telegraf.
3.  I can connect to the Red Lion server from KepWare and UAExpert, using the same connection url: opc.tcp://10.79.116.248:4840, and I am able to read data tags.

### Expected behavior:
## This is the behavior when running gopcua v0.1.13 monitoring example.  I include this to illustrate that the library is able to connect to the Red Lion server.

jnangle-31556s:opcua@v0.1.13 jnangle$ go run examples/monitor/monitor.go -endpoint opc.tcp://10.79.116.248:4840 -node 'ns=2;i=1030'
2021/07/27 13:30:01 *http://opcfoundation.org/UA/SecurityPolicy#NoneMessageSecurityModeNone
2021/07/27 13:30:01 [channel ] sub=4713 ts=2021-07-27T19:28:23Z node=ns=2;i=1030 value=44
2021/07/27 13:30:01 [callback] sub=4714 ts=2021-07-27T19:28:23Z node=ns=2;i=1030 value=44
2021/07/27 13:30:01 [channel ] sub=4713 ts=2021-07-27T19:28:23Z node=ns=2;i=1030 value=44.20000076293945
^Z
[5]+  Stopped                 go run examples/monitor/monitor.go -endpoint opc.tcp://10.79.116.248:4840 -node 'ns=2;i=1030'
jnangle-31556s:opcua@v0.1.13 jnangle$ 


### Actual behavior:

D:\InfluxDB\telegraf-latest\telegraf-1.20.0>.\telegraf --config http://localhost:8086/api/v2/telegrafs/07d830630e2b0000 --test
2021-07-27T22:05:50Z I! Starting Telegraf
2021-07-27T22:05:50Z D! [agent] Initializing plugins
2021-07-27T22:05:50Z I! Failed to load certificate: open /etc/telegraf/cert.pem: The system cannot find the path specified.
2021-07-27T22:05:50Z D! [agent] Starting service inputs
2021-07-27T22:06:11Z E! [inputs.opcua] Error in plugin: registerNodes failed: The operation timed out. StatusBadTimeout (0x800A0000)
2021-07-27T22:06:11Z D! [agent] Stopping service inputs
2021-07-27T22:06:11Z D! [agent] Input channel closed
2021-07-27T22:06:11Z D! [agent] Stopped Successfully
2021-07-27T22:06:11Z E! [telegraf] Error running agent: input plugins recorded 1 errors

D:\InfluxDB\telegraf-latest\telegraf-1.20.0>

### Additional info:

<!-- Include gist of relevant config, logs, etc. -->

@srebhan ,

I think I left this comment in the comments for PR9524 - more coffee this morning, I guess.

The code errored out again at the 1-hr mark with the following messages:

```2021-08-03T16:00:07Z D! [outputs.influxdb_v2] Wrote batch of 80 metrics in 3.0625ms
2021-08-03T16:00:07Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 10000 metrics
2021-08-03T16:00:16Z E! [inputs.opcua] Error in plugin: RegisterNodes Read failed: The operation could not complete because the client is not connected to the server. StatusBadServerNotConnected (0x800D0000)
panic: close of closed channel
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x32915ad]

goroutine 7715 [running]:
github.com/gopcua/opcua/uacp.(*Conn).Close(0x0, 0x0, 0x0)
        /go/pkg/mod/github.com/gopcua/opcua@v0.2.0-rc2.0.20210409063412-baabb9b14fd2/uacp/conn.go:166 +0x2d
panic(0x5138100, 0x62858b0)
        /usr/local/go/src/runtime/panic.go:965 +0x1c7
github.com/gopcua/opcua.(*Client).Close(0xc000ec61c0, 0x0, 0x0)
        /go/pkg/mod/github.com/gopcua/opcua@v0.2.0-rc2.0.20210409063412-baabb9b14fd2/client.go:516 +0xe9
github.com/influxdata/telegraf/plugins/inputs/opcua.Connect(0xc00016c340, 0x0, 0x0)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/opcua/opcua_client.go:409 +0x52e
github.com/influxdata/telegraf/plugins/inputs/opcua.(*OpcUA).Gather(0xc00016c340, 0x6414658, 0xc000a3a240, 0xdb5b55, 0xc001387020)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/opcua/opcua_client.go:522 +0x5c5
github.com/influxdata/telegraf/models.(*RunningInput).Gather(0xc00013c690, 0x6414658, 0xc000a3a240, 0xdf661d, 0x5cc0b80)
        /go/src/github.com/influxdata/telegraf/models/running_input.go:117 +0x74
github.com/influxdata/telegraf/agent.(*Agent).gatherOnce.func1(0xc000750de0, 0xc00013c690, 0x6414658, 0xc000a3a240)
        /go/src/github.com/influxdata/telegraf/agent/agent.go:469 +0x46
created by github.com/influxdata/telegraf/agent.(*Agent).gatherOnce
        /go/src/github.com/influxdata/telegraf/agent/agent.go:468 +0xc5

C:\Users\scada_read\Desktop\Installers\InfluxDB\telegraf-1.20.0 3>```

--- a/go.mod
+++ b/go.mod
@@ -121,7 +121,7 @@ require (
 	github.com/google/uuid v1.2.0 // indirect
 	github.com/googleapis/gax-go/v2 v2.0.5 // indirect
 	github.com/googleapis/gnostic v0.4.1 // indirect
-	github.com/gopcua/opcua v0.1.13
+	github.com/gopcua/opcua v0.2.0-rc2.0.20210409063412-baabb9b14fd2
 	github.com/gorilla/mux v1.7.3
 	github.com/gorilla/websocket v1.4.2
 	github.com/gosnmp/gosnmp v1.32.0

--- a/plugins/inputs/opcua/opcua_client.go
+++ b/plugins/inputs/opcua/opcua_client.go
@@ -445,8 +445,10 @@ func Connect(o *OpcUA) error {
 }
 
 func (o *OpcUA) setupOptions() error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(o.ConnectTimeout))
+	defer cancel()
 	// Get a list of the endpoints for our target server
-	endpoints, err := opcua.GetEndpoints(o.Endpoint)
+	endpoints, err := opcua.GetEndpoints(ctx, o.Endpoint)
 	if err != nil {
 		return err
 	}

--- a/plugins/inputs/opcua/opcua_client.go
+++ b/plugins/inputs/opcua/opcua_client.go
@@ -406,10 +406,10 @@ func Connect(o *OpcUA) error {
 		o.state = Connecting
 
 		if o.client != nil {
-			if err := o.client.CloseSession(); err != nil {
+			if err := o.client.Close(); err != nil {
 				// Only log the error but to not bail-out here as this prevents
 				// reconnections for multiple parties (see e.g. #9523).
-				o.Log.Errorf("Closing session failed: %v", err)
+				o.Log.Errorf("Closing connection failed: %v", err)
 			}
 		}
 

